lib/serviceAccount.ts

```diff
@@ -0,0 +1,270 @@
+import crypto from "crypto";
+import { prisma } from "@lib/integration/prismaConnector";
+import { Prisma } from "@prisma/client";
+import { logMessage } from "@lib/logger";
+import { logEvent } from "@lib/auditLogs";
+import { authCheckAndThrow } from "@lib/actions";
+import { checkUserHasTemplateOwnership } from "@lib/templates";
+import { getZitadelClient } from "@lib/integration/zitadelConnector";
+
+const createMachineUser = async (templateId: string) => {
+  const zitadel = await getZitadelClient();
+  const { userId } = await zitadel
+    .addMachineUser({
+      userName: templateId,
+      name: templateId,
+      description: `API Service account for form ${templateId}`,
+      // Access Token Type 1 is JWT
+      accessTokenType: 1,
+    })
+    .catch((err) => {
+      logMessage.error(err);
+      throw new Error("Could not create User on Identity Provider");
+    });
+
+  logMessage.debug(`Service User ID: ${userId} `);
```

Comment:
Do you want to keep the debug logs for now?

Reply:
I was thinking Yet for some... I'll clean them up.